fe/fe-core/src/main/java/org/apache/doris/cloud/storage/DefaultRemote.java

```diff
@@ -190,6 +211,97 @@ public void putObject(File file, String key) throws DdlException {
+        initClient();
+        initPool();
+        // create multipart upload
+        CreateMultipartUploadRequest createMultipartUploadRequest = CreateMultipartUploadRequest.builder()
+                .bucket(obj.getBucket()).key(key).build();
+        CreateMultipartUploadResponse multipartUpload = s3Client.createMultipartUpload(
+                createMultipartUploadRequest);
+        String uploadId = multipartUpload.uploadId();
+
+        // calculate part size
+        long partSize = Config.multi_part_upload_part_size_in_bytes;
+        if (partSize * MULTI_PART_UPLOAD_MAX_PART_NUM < fileSize) {
+            partSize = (fileSize + MULTI_PART_UPLOAD_MAX_PART_NUM - 1) / MULTI_PART_UPLOAD_MAX_PART_NUM;
+        }
+        int totalPartNum = (int) (fileSize / partSize) + (fileSize % partSize == 0 ? 0 : 1);
+        LOG.info("multi part upload file: {}, size: {}, part size: {}, total part num: {}",
+                file.getAbsolutePath(), fileSize, partSize, totalPartNum);
+
+        try (FileInputStream inputStream = FileUtils.openInputStream(file)) {
+            List<CompletedPart> parts = new ArrayList<>();
+            CountDownLatch latch = new CountDownLatch(totalPartNum);
+            int partNum = 1;
+            long totalUploaded = 0;
+            AtomicBoolean failed = new AtomicBoolean(false);
+
+            while (totalUploaded < fileSize && !failed.get()) {
+                long nextPartSize = Math.min(partSize, fileSize - totalUploaded);
+                int partNumConst = partNum;
+                POOL.submit(() -> {
+                    if (failed.get()) {
+                        return;
+                    }
+                    LOG.debug("start multi part upload for num: {}", partNumConst);
+                    UploadPartRequest uploadPartRequest = UploadPartRequest.builder().bucket(obj.getBucket()).key(key)
+                            .uploadId(uploadId).partNumber(partNumConst).build();
+                    try {
+                        UploadPartResponse uploadPartResponse = s3Client.uploadPart(uploadPartRequest,
+                                RequestBody.fromInputStream(inputStream, nextPartSize));
+                        synchronized (parts) {
+                            parts.add(CompletedPart.builder().partNumber(partNumConst).eTag(uploadPartResponse.eTag())
+                                    .build());
+                        }
+                        LOG.debug("finish multi part upload for num: {}", partNumConst);
+                    } catch (Exception e) {
+                        LOG.warn("Failed to multi part upload for num: {}", partNumConst, e);
+                        failed.set(true);
+                    } finally {
+                        latch.countDown();
+                    }
+                });
+                totalUploaded += nextPartSize;
+                partNum++;
+            }
+            for (int i = 0; i < Config.multi_part_upload_max_seconds / 10; i++) {
```

Comment:
LOG more details here, how long it has been waiting
it seems does not make sense  i < config.multi...seconds / 10, however latch wait interval is 10 sec 

we may need to accumulate time waited by the latch and check if the sum exceeds Config.multi_part_upload_max_seconds